File: my-folder/1031-add-to-array-form-of-integer/solution.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> ans = new ArrayList<>();
        int x = num.length-1;
        int carry = k;
        while (x >= 0 && carry > 0) {
            int sum = carry + num[x];
            num[x] =  sum % 10;
            carry = sum / 10;
            x--;

        }
        List<Integer> num2 = new ArrayList<>();

        for (int i = 0; i < num.length; i++) {
            num2.add(num[i]);
        }

        if (carry > 0) {
            while (carry > 0) {
                num2.add(0, carry % 10);
                carry /= 10;
            }
        }


        return num2;
    }
}

File: my-folder/1031-add-to-array-form-of-integer/solution.cpp
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        int carry=0;
        for (int i = num.size()-1; i>=0 || k>0 || carry>0; i--) {
            if(k==0 && carry==0) break;
            int sum=0;
            int x=i>=0?num[i]:0;
            sum=x+k%10+carry;
            carry=sum/10;
            if(i>=0){
                num[i]=sum%10;
            }
            else{
                num.emplace(num.begin(),sum%10);
            }
            k/=10;
        }

        return num;
    }
};
